internal/coordinator/coordinator_test.go

```diff
@@ -0,0 +1,1043 @@
+	"github.com/FoundationDB/fdb-kubernetes-operator/internal"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Change coordinators", func() {
+	var cluster *fdbv1beta2.FoundationDBCluster
+	var adminClient *mock.AdminClient
+
+	BeforeEach(func() {
+		cluster = internal.CreateDefaultCluster()
+		cluster.Spec.ProcessCounts = fdbv1beta2.ProcessCounts{
+			Storage: 4,
+		}
+		cluster.Spec.CoordinatorSelection = []fdbv1beta2.CoordinatorSelectionSetting{
+			{
+				ProcessClass: fdbv1beta2.ProcessClassStorage,
+				Priority:     math.MaxInt32,
+			},
+			{
+				ProcessClass: fdbv1beta2.ProcessClassLog,
+				Priority:     0,
+			},
+		}
+		Expect(internal.SetupClusterForTest(cluster, k8sClient)).NotTo(HaveOccurred())
+		Expect(k8sClient.Create(context.Background(), cluster)).NotTo(HaveOccurred())
+
+		var err error
+		adminClient, err = mock.NewMockAdminClientUncast(cluster, k8sClient)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Describe("selectCoordinators", func() {
+		Context("with a single FDB cluster", func() {
+			var status *fdbv1beta2.FoundationDBStatus
+			var candidates []locality.Info
+
+			JustBeforeEach(func() {
+				var err error
+				status, err = adminClient.GetStatus()
+				Expect(err).NotTo(HaveOccurred())
+
+				candidates, err = selectCoordinatorsLocalities(logr.Discard(), cluster, status)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			When("all processes are healthy", func() {
+				It("should only select storage processes", func() {
+					Expect(cluster.DesiredCoordinatorCount()).To(BeNumerically("==", 3))
+					Expect(candidates).To(HaveLen(cluster.DesiredCoordinatorCount()))
+
+					// Only select Storage processes since we select 3 processes and we have 4 storage processes
+					for _, candidate := range candidates {
+						Expect(candidate.Class).To(Equal(fdbv1beta2.ProcessClassStorage))
+					}
+				})
+			})
+
+			When("when one storage process is marked for removal", func() {
+				var removedProcess fdbv1beta2.ProcessGroupID
+
+				BeforeEach(func() {
+					removedProcess = internal.PickProcessGroups(cluster, fdbv1beta2.ProcessClassStorage, 1)[0].ProcessGroupID
+					cluster.Spec.ProcessGroupsToRemove = []fdbv1beta2.ProcessGroupID{
+						removedProcess,
+					}
+					Expect(cluster.ProcessGroupIsBeingRemoved(removedProcess)).To(BeTrue())
+				})
+
+				It("should only select storage processes and exclude the removed process", func() {
+					Expect(cluster.DesiredCoordinatorCount()).To(BeNumerically("==", 3))
+					Expect(candidates).To(HaveLen(cluster.DesiredCoordinatorCount()))
+
+					// Only select Storage processes since we select 3 processes and we have 4 storage processes
+					for _, candidate := range candidates {
+						Expect(candidate.ID).NotTo(Equal(removedProcess))
+						Expect(candidate.Class).To(Equal(fdbv1beta2.ProcessClassStorage))
+					}
+				})
+			})
+
+			//When("when one storage process is excluded", func() {
```

Comment:
again, would be good to delete or uncomment - though ideally this could be adjusted to still pass